lib/system/setops.nim

```diff
@@ -9,6 +9,17 @@ func incl*[T](x: var set[T], y: T) {.magic: "Incl".} =
     a.incl(4)
     assert a == {1, 2, 3, 4, 5}
 
+func toSet*[char|byte|bool|int16|uint16|enum](ar: openArray[T]): set[T] =
+  ##Iterates through an openArray making a set from it.
```

Comment:
just like sequtils.toSeq, you should allow the more general case of input iterator which subsubmes the `openArray` case.
this'll allow:
```nim
doAssert toSet(someIter()) == {'a', 'b'}
doAssert toSet("ab") == {'a', 'b'}
doAssert toSet(['a','a']) == {'a'}
```
etc